### PR TITLE
core: enable async stacks in the DevtoolsLog gatherer

### PR DESCRIPTION
**Pull request by adamraine**
Alternative to https://github.com/GoogleChrome/lighthouse/pull/14716

From what I can tell, we only access the initiator information when looking at network records constructed from the DevtoolsLog. I think it makes sense to encapsulate this with the DevtoolsLog.

### CODE REVIEW

File: core/gather/gatherers/devtools-log.js [hunk 1]
```diff
@@ -28,12 +30,56 @@ class DevtoolsLog extends FRGatherer {
...
+
+    this._disableAsyncStacks = () => {};
+  }
+
+  /**
+   * Enables `Debugger` domain to receive async stacktrace information on network request initiators.
+   * This is critical for tracking attribution of tasks and performance simulation accuracy.
+   * @param {LH.Gatherer.FRProtocolSession} session
+   */
+  async _enableAsyncStacks(session) {
```
Comment (brendankenny) [inline core/gather/gatherers/devtools-log.js:42]: Looks like this doesn't get enabled for legacy, though? If so, kind of worried we don't have any smoke test coverage of async stack trace 😬

Comment (adamraine) [reply core/gather/gatherers/devtools-log.js:42]: It does get called in legacy. `startSensitiveInstrumentation` is run in `beforePass` by default:

https://github.com/GoogleChrome/lighthouse/blob/4c5c65e256591ade1381872e7274e7f94ff33be2/core/gather/base-gatherer.js#L75-L83

Comment (brendankenny) [reply core/gather/gatherers/devtools-log.js:42]: I don't think the DevtoolsLog gatherer is run in legacy though?

Comment (adamraine) [reply core/gather/gatherers/devtools-log.js:42]: Ah your right, we would need to do this in legacy driver :/

This is indeed concerning for lack of smoke tests

Comment (adamraine) [reply core/gather/gatherers/devtools-log.js]: I've added a smoke test for this. I tested and dbw now fails if async stacks aren't enabled.

**Comment by adamraine** [comment]
This PR has become a bit of a rabbit hole. While I do think it would be good to land this in the long term, it opens the door for some more complex challenges relating to scripts gathering and third parties.

Calling `Debugger.enable` when the domain is disabled will flush any `Debugger.scriptParsed` events that the session hadn't emitted yet. Since this only happens when the domain was in the disabled state, the events are emitted on the *first* invocation of `Debugger.enable` which previously happened in `prepare.js` as part of our navigation/timespan prework.

This PR moves that invocation into the `DevtoolsLog` gatherer, which moves the *new* first invocation of `Debugger.enable` into the `Scripts` gatherer. This doesn't mean anything for navigation mode, since the domain will always be enabled while we are on `about:blank` before the navigation starts. However, timespan mode can start on an already loaded page, which means there could be `Debugger.scriptParsed` events waiting to emit that will now be caught by the `Scripts` gatherer since the first invocation of `Debugger.enable` happens there. This means timespan mode will collect all scripts that are loaded during the timespan, *as well as* any scripts that were on the page at the start of the timespan. In theory this is a better way of collecting scripts for an arbitrary timespan however...

With the introduction of the new 3P system, we will throw an error if we encounter a url that doesn't exist in the network records. Now we are collecting scripts that weren't fetched during the timespan, [which can cause this error to be thrown](https://github.com/GoogleChrome/lighthouse/actions/runs/4018736614/jobs/6904816627) if we attempt to use a url from one of the pre-timespan scripts in say the `valid-source-map` audit:

https://github.com/GoogleChrome/lighthouse/blob/6de2ed25cdf2da46ed1bc261486d638f7a5adc49/core/audits/valid-source-maps.js#L56-L62

In the future, I imagine we will classify the urls in `Scripts` but for now I think the easiest workaround is to call `Debugger.enable` in the navigation/timespan prework so all the `Debugger.scriptParsed` events are flushed before gathering starts.

**Comment by adamraine** [comment]
It has some organizational benefits, but I don't see myself spending any time on it in the near future

**Pull request closed without merging**